tensorflow_datasets/image/coco.py

```diff
@@ -262,6 +264,413 @@ def build_bbox(x, y, width, height):
+    """Returns SplitGenerators."""
+    root_url = "http://images.cocodataset.org/"
+    urls = {
+        # Train/validation set
+        "train_images": "zips/train2017.zip",
+        "val_images": "zips/val2017.zip",
+        "panoptic_annotations_trainval2017": "annotations/panoptic_annotations_trainval2017.zip",
+    }
+    extracted_paths = dl_manager.download_and_extract({
+        key: root_url + url for key, url in urls.items()
+    })
+
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            num_shards=10,
+            gen_kwargs=dict(
+                image_dir=extracted_paths["train_images"],
+                annotation_dir=extracted_paths["panoptic_annotations_trainval2017"],
+                split_type="train2017",
+            )),
+        tfds.core.SplitGenerator(
+            name=tfds.Split.VALIDATION,
+            num_shards=10,
+            gen_kwargs=dict(
+                image_dir=extracted_paths["val_images"],
+                annotation_dir=extracted_paths["panoptic_annotations_trainval2017"],
+                split_type="val2017",
+            )),
+    ]
+
+  def _generate_examples(
+      self, image_dir, annotation_dir, split_type):
+    """Generate examples as dicts.
+
+    Args:
+      image_dir: `str`, directory containing the images
+      annotation_dir: `str`, directory containing panoptic annotations
+      split_type: `str`, <split_name><year> (ex: train2017)
+
+    Yields:
+      Generator yielding the next samples
+    """
+    instance_filename = "panoptic_{}.json"
+
+    # Load the label names and images
+    instance_path = os.path.join(
+        annotation_dir,
+        "annotations",
+        instance_filename.format(split_type),
+    )
+    coco_annotation = CocoPanopticAnnotation(instance_path)
+    # Each category is a dict:
+    # {
+    #    'id': 5,
+    #    'isthing': 1  
+    #    'name': 'airplane',
+    #    'supercategory': 'vehicle',
+    # }
+    categories = coco_annotation.categories
+    # Each image is a dict:
+    # {
+    #     'id': 139,
+    #     'file_name': '000000397133.jpg',
+    #     'flickr_url': 'http://farm9.staticflickr.com/8035/8024364858_9c41dc1666_z.jpg',
+    #     'coco_url': 'http://images.cocodataset.org/val2017/000000000139.jpg',
+    #     'license': 2,
+    #     'date_captured': '2013-11-21 01:34:01',
+    #     'height': 426,
+    #     'width': 640,
+    # }
+    images = coco_annotation.images
+
+    self.info.features["objects"]["label"].names = [
+        c["name"] for c in categories
+    ]
+    
+    categories_id2name = {c["id"]: c["name"] for c in categories}
+    
+    # Extract the panoptic_{}.zip folder containing the png images
+    panoptic_image_filename = "panoptic_{}.zip"
+    panoptic_image_path = os.path.join(
+        annotation_dir,
+        "annotations",
+        panoptic_image_filename.format(split_type)
+    )
+    with ZipFile(panoptic_image_path) as f:
```

Comment:
Why didn't use `dl_manager.extract` method?

Reply:
I got something similar [here](https://github.com/tensorflow/datasets/blob/813b5d2aa9c8c1a4c8d0aae7130239babe276983/tensorflow_datasets/image/imagenet.py#L138).

Reply:
@Conchylicultor @pierrot0 what's the difference of `zipfile` package and `dl_manager`?

Reply:
The problem here is that ZipFile will not use tf.gfile, which it should.
Using dl_manager will do the right thing (open the file with tf.gfile), and use the right format if not zip.

So I would prefer you to use dl_manager Chanchal.
Otherwise Zipfile is fine if code is not a repeat of the extractor and you know it's alway a zip.

Reply:
Actually, the file is always a `zip` file. So, should I change ?

Reply:
Yes, as Pierre say, `ZipFile` won't works for non-standard file systems (ex: cannot read from GCS & cie), so it is better to use `dl_manager.extract()`.

Additionally, `dl_manager` will cache and reuse the result, while using `ZipFile` will re-extract the data or may crash due to data already has been extracted once.